polygeist/lib/Dialect/Polygeist/Transforms/LICM.cpp

```diff
@@ -896,7 +905,7 @@ class SYCLAccessorVersionBuilder : public SCFLoopVersionBuilder {
     return createSYCLAccessorSubscriptOp(accessor, id, builder, loc);
   }
 
-  std::pair<TypedValue<MemRefType>, TypedValue<MemRefType>> AccessorPair;
+  const SmallVector<AccessorPairType> &accessorPairs;
```

Comment:
If you don't need ownership here, use `ArrayRef` instead

Reply:
Good idea.